src/map.cpp

```diff
@@ -4528,40 +4438,56 @@ template<typename T>
 void map::process_items_in_vehicle( vehicle &cur_veh, submap &current_submap,
                                     T processor, std::string const &signal )
 {
-    std::vector<int> cargo_parts = cur_veh.all_parts_with_feature( VPFLAG_CARGO, true );
-    for( int part : cargo_parts ) {
-        process_vehicle_items( cur_veh, part );
-    }
-
     std::vector<int> tanks = cur_veh.all_parts_with_feature( VPFLAG_FLUIDTANK, false );
     for( const int &idx : tanks ) {
         const point partloc = cur_veh.global_pos() + cur_veh.parts[idx].precalc[0];
         const tripoint partpos = tripoint( partloc, abs_sub.z );
         cur_veh.parts[idx].process_contents( partpos );
     }
 
+    std::vector<int> cargo_parts = cur_veh.all_parts_with_feature( VPFLAG_CARGO, true );
+    for( int part : cargo_parts ) {
+        process_vehicle_items( cur_veh, part );
+    }
+
+    const bool heater_is_on = cur_veh.has_part( "HEATER", true ) && cur_veh.engine_on;
+    const point veh_pos = cur_veh.global_pos();
     for( auto &active_item : cur_veh.active_items.get() ) {
         if( cargo_parts.empty() ) {
             return;
         } else if( !cur_veh.active_items.has( active_item ) ) {
             continue;
         }
-
         auto const it = std::find_if( begin( cargo_parts ), end( cargo_parts ), [&]( int const part ) {
             return active_item.location == cur_veh.parts[static_cast<size_t>( part )].mount;
         } );
 
         if( it == std::end( cargo_parts ) ) {
             continue; // Can't find a cargo part matching the active item.
         }
-
+        auto &item_iter = active_item.item_iterator;
         // Find the cargo part and coordinates corresponding to the current active item.
-        auto const part_index = static_cast<size_t>( *it );
-        const point partloc = cur_veh.global_pos() + cur_veh.parts[part_index].precalc[0];
+        const size_t part_index = static_cast<size_t>( *it );
+        const vehicle_part &pt = cur_veh.parts[part_index];
+        const point partloc = veh_pos + pt.precalc[0];
         // TODO: Make this 3D when vehicles know their Z-coordinate
-        const tripoint item_location = tripoint( partloc, abs_sub.z );
+        const tripoint item_loc = tripoint( partloc, abs_sub.z );
+        if( item_iter->is_food() || item_iter->is_food_container() ) {
+            const vpart_info &pti = pt.info();
+            int temp = g->get_temperature( item_loc );
+            if( heater_is_on ) {
+                temp = std::max( temp, COLD_TEMPERATURE + 1 );
```

Comment:
Implementing it as a field would be significantly more difficult and I don't think it's that "incredulous" the whole vehicle would get warmed by the engine heater.  This would be coupled with the fact a larger vehicle should have a larger engine and thereby a larger amount of heat output.  Then you have to talk about how the heater is part of the engine and what you're suggesting would actually require a whole "vent dynamic" for very little pay off.

I can take the heater out if you feel nothing is better than this, but those are the options here.

Reply:
To follow-up regarding #25440, what's being requested there is a heater implemented as a field.  This will not work as a vehicle part for the same reason smoke out of a muffler is buggy af, meaning the field is local to the map, not the vehicle.  None of the field processing is equipped to use a vehicle as a frame of reference.

If it was done as a furniture item, it would be possible to build one on a map and emit heat, but there's not an easy way to make this a vehicle part, aside from the solution I have here.